darts/models/regression_model.py

```diff
@@ -39,263 +51,226 @@ def __init__(self,
+        # turning lags into array of int or None
+        self.lags, self.lags_covariates = _process_lags(lags, lags_covariates)
 
-        Parameters
-        ----------
-        series : TimeSeries
-            TimeSeries object containing the target values.
-        exog : TimeSeries, optional
-            TimeSeries object containing the exogenous values.
-        """
-        super().fit(series, exog)
-        raise_if(exog is not None and self.lags_exog is None,
-            "`exog` not None in `fit()` method call, but `lags_exog` is None in constructor. "
-        )
-        raise_if(exog is None and self.lags_exog is not None,
-            "`exog` is None in `fit()` method call, but `lags_exog` is not None in constructor. "
-        )
-        self.exog_columns = exog.columns.values.tolist() if exog is not None else None
-        self.nr_exog = exog.width*len(self.lags_exog) if exog is not None else 0
-
-        # Rename column if necessary
-        target_column = str(series.columns[0])
-        if self.exog_columns is not None and target_column in self.exog_columns:
-            series = TimeSeries.from_dataframe(series.pd_dataframe().rename({target_column: "Target"}))
-        self.target_column = series.columns[0]
-
-        # Prepare data
-        training_x = self._create_training_data(series=series, exog=exog)
-        training_y = series[training_x.time_index]
-
-        # Fit model
-        if "series" in signature(self.model.fit).parameters:
-            self.model.fit(
-                series=training_y,
-                exog=training_x,
-                **kwargs
-            )
+        if self.lags is not None:
+            self.lags_indices = np.array(self.lags) * (-1)
         else:
-            self.model.fit(
-                training_x.pd_dataframe(),
-                training_y.pd_dataframe().values.ravel(),
-                **kwargs
-            )
+            self.lags_indices = None
 
-        if self.max_lag == 0:
-            self.prediction_data = pd.DataFrame(columns=series.stack(other=exog).columns)
-        elif exog is not None:
-            self.prediction_data = series.stack(other=exog)[-self.max_lag:]
+        if self.lags_covariates is not None:
+            if 0 in self.lags_covariates:
+                self.cov_lags_indices = (np.array(self.lags_covariates) + 1) * (-1)
+            else:
+                self.cov_lags_indices = (np.array(self.lags_covariates)) * (-1)
         else:
-            self.prediction_data = series[-self.max_lag:]
+            self.cov_lags_indices = None
 
-        self._fit_called = True
-
-    def _create_training_data(self, series: TimeSeries, exog: TimeSeries = None):
-        """ Create dataframe of exogenous and endogenous variables containing lagged values.
+    def fit(
+        self,
+        series: Union[TimeSeries, Sequence[TimeSeries]],
+        covariates: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
```

Comment:
Awesome! I'll try taking a look at your PR and adapting accordingly since I see that, for example, the current `SimpleInferenceDataset` will be replaced🤔 And yep, I guess that in that way I can remove some tricks I'm adopting right now to handle the 0 index covariate 👍🏼

Reply:
Cool! Just so you know: I'm still working on it, not done yet. I still need a bit of time because it's quite a big change. But I'm hopeful that the datasets will stay more or less as they are.